### PR TITLE
Deavina_Assignment_17_Done_40min

### PR DESCRIPTION
**Pull request by DeavinaSJun20**
image
image

### CODE REVIEW

File: src/deavinaSJun20/TestCount.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+		int countDigit = 0, countUppercase = 0, countLowercase = 0, countSpecialchar = 0, countLetter = 0;
+		for (int i = 0; i < name.length(); i++) {
+			if (Character.isDigit(name.charAt(i))) {
+				countDigit++;
+			} else if (Character.isLetter(name.charAt(i))) {
+				countLetter++;
+				if (Character.isLowerCase(name.charAt(i))) {
+					countLowercase++;
+				} else if (Character.isUpperCase(name.charAt(i))) {
+					countUppercase++;
```
Comment (KrishnaKTechnocredits) [inline src/deavinaSJun20/TestCount.java]: You can achieve the output without using countLetter variable and else if (Character.isLetter(name.charAt(i))) block just think about it.

File: src/deavinaSJun20/TestIndexCount.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+	public static void main(String[] args) {
+		String name="technocredits";
+		System.out.println("Entered String : "+name);
+		new TestIndexCount().repeatedCharIndex(name);
+	}
+	//function to print repeated characters
+	public void repeatedCharIndex(String name) {
+		for(int i=0;i<name.length();i++) {
+			char ch=name.charAt(i);
+			if(i==name.indexOf(ch) && i!=name.lastIndexOf(ch))
```
Comment (KrishnaKTechnocredits) [inline src/deavinaSJun20/TestIndexCount.java:13]: perfect logic.

File: src/deavinaSJun20/TestCount.java (hunk 1 above)
Comment (DeavinaSJun20) [reply src/deavinaSJun20/TestCount.java]: review comments incorporated

File: src/deavinaSJun20/TestCount.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+			} else if (Character.isLowerCase(name.charAt(i))) {
+				countLowercase++;
+			} else if (Character.isUpperCase(name.charAt(i))) {
+				countUppercase++;
+			} else {
+				countSpecialchar++;
+			}
+		}
+		System.out.println("Total Digits in String : " + countDigit);
+		System.out.println("Total Letters in String : " + (countUppercase + countLowercase));
```
Comment (KrishnaKTechnocredits) [inline src/deavinaSJun20/TestCount.java:34]: Good job, Perfect logic.

**Pull request merged**